### PR TITLE
feat: install shots in albums and select mode

### PR DESCRIPTION
**Pull request by pvinis**
### Description

This PR resolves [MOPLAT-340]

Install shots in albums and select mode

⚠️  after this PR is merged and shipped, we will need to reinstall the app, because there was some change in the GlobalStore models, and we don't have migrations yet.

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-340]: https://artsyproduct.atlassian.net/browse/MOPLAT-340?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/ambient.d.ts [hunk 1]
```diff
@@ -4,5 +4,3 @@ declare module "" {
     const __STORYBOOK__: boolean
   }
 }
-
-declare function assertNever(val: never): void
```
Comment (pvinis) [inline src/ambient.d.ts:8]: this was not enough for TS to figure out types well, so I moved it to an actual function.

File: src/app/screens/Dev/DevMenu.tsx [hunk 2]
```diff
@@ -70,8 +65,8 @@ export const DevMenu = () => {
                 const album = {
                   name: nativeAlbum.name,
                   artworkIds: nativeAlbum.artworkIDs,
-                  documentIds: [],
                   installShotUrls: [],
+                  documentIds: [],
```
Comment (pvinis) [inline src/app/screens/Dev/DevMenu.tsx:69]: just keeping the right order

File: src/app/store/Models/SelectModeModel.ts [hunk 3]
```diff
@@ -1,39 +1,89 @@
-import { action, Action } from "easy-peasy"
+import { action, Action, computed, Thunk, Computed, thunk } from "easy-peasy"
+import { assertNever } from "shared/utils/assertNever"
 
-type ItemType = "works" | "documents"
+type ItemType = "artwork" | "install" | "document"
```
Comment (pvinis) [inline src/app/store/Models/SelectModeModel.ts:4]: adding install type

File: src/app/store/Models/SelectModeModel.ts [hunk 4]
```diff
@@ -1,39 +1,89 @@
...
 
 export interface SelectModeModel {
+  // state
   isSelectModeActive: boolean
-  items: Record<ItemType, string[]>
-  toggleSelectMode: Action<this>
-  cancelSelectMode: Action<this>
-  selectItem: Action<this, { itemType: ItemType; item: string }>
-  selectAllItems: Action<this, { itemType: ItemType; allItems: string[] }>
+  artworks: Array<string>
```
Comment (pvinis) [inline src/app/store/Models/SelectModeModel.ts:9]: separate these out, and have `items` as a computed one.

i noticed in most cases we care about specific types of items in the app code, and whenever we actually used something like a total `items` thing, it was just for counting, so having the base state be the 3 arrays makes sense.

File: src/app/store/Models/SelectModeModel.ts [hunk 5]
```diff
@@ -1,39 +1,89 @@
...
+      arrayToLookAt = "installs"
+      break
+    case "document":
+      arrayToLookAt = "documents"
+      break
+    default:
+      assertNever(type)
+  }
+  return arrayToLookAt
+}
```
Comment (rajsam003) [inline src/app/store/Models/SelectModeModel.ts:89]: 👏

**Review by rajsam003 — APPROVED** [review]

**Comment by rajsam003** [comment]
Looks good. Feel free to merge

**Pull request merged**